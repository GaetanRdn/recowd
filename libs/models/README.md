# models

## description

Contains all models defined in the workspace.
