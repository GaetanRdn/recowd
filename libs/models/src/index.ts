export * from './ui/contract-type';
export * from './ui/education';
export * from './ui/experience';
export * from './ui/partner';
export * from './ui/role';
