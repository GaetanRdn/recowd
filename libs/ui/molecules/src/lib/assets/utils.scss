@function map-get-deep($map, $level, $subLevel) {
  @return map-get($map, $level, $subLevel);
}

@mixin use-typography($typography) {
  @each $key in font-size, font-family, font-weight, line-height {
    #{$key}: map-get($typography, $key);
  }
}
