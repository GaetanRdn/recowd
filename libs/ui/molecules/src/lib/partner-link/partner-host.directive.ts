import { Directive, ViewContainerRef } from '@angular/core';

@Directive({
  selector: '[rcHost]',
  standalone: true,
})
export class PartnerHostDirective {
  constructor(public viewContainerRef: ViewContainerRef) {}
}
