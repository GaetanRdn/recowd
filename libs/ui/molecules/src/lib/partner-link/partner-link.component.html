<a [href]="partner.link" rcButtonIcon target="_blank">
  <ng-template rcHost></ng-template>
</a>
