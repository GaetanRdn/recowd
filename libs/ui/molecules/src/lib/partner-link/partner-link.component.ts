import {
  ChangeDetectionStrategy,
  Component,
  Input,
  OnInit,
  ViewChild,
} from '@angular/core';
import { ButtonDirective, CustomIconComponent } from '@recowd/ui-atoms';
import { Partner } from '@recowd/models';
import { Required } from '@recowd/utility-types';
import { PartnerHostDirective } from './partner-host.directive';
import { NorsysCustomIconComponent } from '../custom-icons/norsys-custom-icon/norsys-custom-icon.component';
import { CbpCustomIconComponent } from '../custom-icons/cbp-custom-icon/cbp-custom-icon.component';

@Component({
  selector: 'rc-partner-link',
  standalone: true,
  imports: [ButtonDirective, PartnerHostDirective],
  templateUrl: './partner-link.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class PartnerLinkComponent implements OnInit {
  @Input() @Required() public partner!: Partner;

  @ViewChild(PartnerHostDirective, { static: true })
  private readonly container!: PartnerHostDirective;

  public ngOnInit(): void {
    this.container.viewContainerRef.clear();
    this.container.viewContainerRef.createComponent<CustomIconComponent>(
      this.partner.icon === 'NorsysCustomIconComponent'
        ? NorsysCustomIconComponent
        : CbpCustomIconComponent
    );
  }
}
