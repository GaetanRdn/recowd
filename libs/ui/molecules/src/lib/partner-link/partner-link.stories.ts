import { Args, Meta, moduleMetadata, Story } from '@storybook/angular';
import { PartnerLinkComponent } from './partner-link.component';
import { NorsysCustomIconComponent } from '../custom-icons/norsys-custom-icon/norsys-custom-icon.component';
import { CbpCustomIconComponent } from '../custom-icons/cbp-custom-icon/cbp-custom-icon.component';

export default {
  title: 'molecules/partner-link',
  parameters: {
    jest: 'partner-link',
  },
  decorators: [
    moduleMetadata({
      imports: [
        PartnerLinkComponent,
        NorsysCustomIconComponent,
        CbpCustomIconComponent,
      ],
    }),
  ],
  argTypes: {
    icon: {
      control: {
        options: [NorsysCustomIconComponent, CbpCustomIconComponent],
      },
    },
  },
} as Meta<
  PartnerLinkComponent<CbpCustomIconComponent | NorsysCustomIconComponent>
>;

const Template: Story = (args: Args) => ({
  props: {
    partner: {
      link: args['link'],
      icon: args['icon'],
    },
  },
  template: `<rc-partner-link [partner]="partner"></rc-partner-link>`,
});

export const Default = Template.bind({});
Default.args = {
  link: 'https://www.norsys.fr/',
  icon: NorsysCustomIconComponent,
};
