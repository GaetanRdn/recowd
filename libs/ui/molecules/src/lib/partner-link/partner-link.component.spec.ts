import { TestBed, waitForAsync } from '@angular/core/testing';
import { PartnerLinkComponent } from './partner-link.component';
import { Component } from '@angular/core';
import { TemplateLookup } from '@recowd/test/utils';
import { Partner } from '@recowd/models';

describe('PartnerLinkComponent', () => {
  let templateLookup: TemplateLookup<HostComponent>;

  beforeEach(waitForAsync(() => {
    TestBed.configureTestingModule({
      declarations: [HostComponent],
      imports: [PartnerLinkComponent],
    }).compileComponents();

    templateLookup = new TemplateLookup(HostComponent);
  }));

  it.each([
    { link: 'www.test.fr', icon: 'CbpCustomIconComponent' },
    {
      link: 'www.paulo.com',
      icon: 'NorsysCustomIconComponent',
    },
  ])('should create with %j', (partner: Partner) => {
    templateLookup.hostComponent.partner = partner;

    templateLookup.detectChanges();

    expect(templateLookup.firstChildElement).toMatchSnapshot();
  });
});

@Component({
  template: ` <rc-partner-link [partner]="partner"></rc-partner-link>`,
})
class HostComponent {
  public partner!: PartnerLinkComponent['partner'];
}
