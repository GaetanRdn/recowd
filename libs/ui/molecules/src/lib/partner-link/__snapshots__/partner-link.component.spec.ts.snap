// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PartnerLinkComponent should create with {"link":"www.paulo.com","icon":"NorsysCustomIconComponent"} 1`] = `
<rc-partner-link>
  <a
    class="rc-button-icon"
    href="www.paulo.com"
    rcbuttonicon=""
    target="_blank"
  >
    <rc-norsys-custom-icon>
      <rc-custom-icon>
        <img
          alt="norsys"
          height="300"
          src="assets/images/norsys.png"
          width="300"
        />
      </rc-custom-icon>
    </rc-norsys-custom-icon>
    
  </a>
</rc-partner-link>
`;

exports[`PartnerLinkComponent should create with {"link":"www.test.fr","icon":"CbpCustomIconComponent"} 1`] = `
<rc-partner-link>
  <a
    class="rc-button-icon"
    href="www.test.fr"
    rcbuttonicon=""
    target="_blank"
  >
    <rc-cbp-custom-icon>
      <rc-custom-icon>
        <img
          alt="cbp"
          height="300"
          src="assets/images/cbp.svg"
          width="300"
        />
      </rc-custom-icon>
    </rc-cbp-custom-icon>
    
  </a>
</rc-partner-link>
`;
